packages/did-provider-ebsi/src/functions.ts

```diff
@@ -33,3 +36,30 @@ export function generateEbsiPrivateKeyHex(specInfo?: EbsiDidSpecInfo, privateKey
   }
   return u8a.toString(randomBytes(length), 'base16')
 }
+
+/**
+ * Returns the public key in the correct format to be used with the did registry v5
+ * - in case of Secp256k1 - returns the uncompressed public key as hex string prefixed with 0x04
+ * - in case of Secp256r1 - returns the jwk public key as hex string
+ * @param {{ key: IKey, type: EbsiKeyType }} args
+ *  - key is the cryptographic key containing the public key
+ *  - type is the type of the key which can be Secp256k1 or Secp256r1
+ *  @returns {string} The properly formatted public key
+ *  @throws {Error} If the key type is invalid
+ */
+export const formatEbsiPublicKey = (args: { key: IKey; type: EbsiKeyType }): string => {
+  const { key, type } = args
+  let bytes = getBytes('0x' + key.publicKeyHex, 'key')
+  switch (type) {
+    case 'Secp256k1': {
+      return SigningKey.computePublicKey(bytes, false)
+    }
+    case 'Secp256r1': {
+      const jwk: JsonWebKey = toJwk(key.publicKeyHex, type, { use: JwkKeyUse.Signature, key })
```

Comment:
Why do we need to convert the public key hex to JWK and then back to public key hex?

Reply:
In other words, what is the difference? If there is one, please describe it

Reply:
It's converting the hex key to a JWK object and then to a JSON string. There is no conversion back to hex. What do I miss?

Reply:
Please reread the implementation (which will be wrong anyway, as that is not how you convert a jwk to hex to begin with. We have functions for these conversions). The comment reads it should be hex and the last thing you do is convert to base16 which is hex.

As mentioned why would you need to do it to begin with? If the comment is correct then you already had the hex key, if not then you need to adjust, but converting a jwk to base16 directly always is wrong, you need to get the public key component first (and we have functions for these)

Reply:
Added a comment explaining what and why is happening.